merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/TypescriptCodeGenerationService.java

```diff
@@ -33,4 +171,158 @@ private static String joinLines(final Iterable<String> result) {
   private static String indent(final String s) {
     return joinLines(s.lines().map(line -> "  " + line).toList());
   }
+
+
+  private static String valueSchemaToTypescript(final ValueSchema valueSchema) {
+    return valueSchema.match(new ValueSchema.Visitor<>() {
+      @Override
+      public String onReal() {
+        return "number";
+      }
+
+      @Override
+      public String onInt() {
+        return "number";
+      }
+
+      @Override
+      public String onBoolean() {
+        return "boolean";
+      }
+
+      @Override
+      public String onString() {
+        return "string";
+      }
+
+      @Override
+      public String onDuration() {
+        return "Duration";
+      }
+
+      @Override
+      public String onPath() {
+        return "string";
+      }
+
+      @Override
+      public String onSeries(final ValueSchema value) {
+        return valueSchemaToTypescript(value) + "[]";
+      }
+
+      @Override
+      public String onStruct(final Map<String, ValueSchema> values) {
+        final var result = new StringBuilder("{");
+
+        for (final var member: values.keySet()) {
+          result
+              .append(member)
+              .append(": ")
+              .append(valueSchemaToTypescript(values.get(member)))
+              .append(", ");
+        }
+
+        result.append("}");
+        return result.toString();
+      }
+
+      @Override
+      public String onVariant(final List<ValueSchema.Variant> variants) {
+        final var result = new StringBuilder("(");
+
+        for (final var variant: variants) {
+          result
+              .append(" | \"")
+              .append(variant.label())
+              .append("\"");
+        }
+
+        result.append(")");
+        return result.toString();
+      }
+    });
+  }
+
+  private static String valueSchemaToTypescriptProfile(final ValueSchema valueSchema) {
+    var basicTypescript = valueSchemaToTypescript(valueSchema);
+    if (basicTypescript.equals("number")) {
+      return "Real";
+    } else {
+      return "Discrete<" + basicTypescript + ">";
+    }
+  }
+
+  private static String valueSchemaToTypescriptDefault(final ValueSchema valueSchema) {
```

Comment:
Are these defaults intended to be replaced by user-defined values? Only asking because I find the defaults to be a little unexpected. For example:
- `"string"` instead of `""` as the string default, and
- `/path` instead of something like `/dev/null` as the path default.

Reply:
No, they values of the defaults don't matter at all. They are never used or changed, and are discarded by the end :). In any other situation these values would be dead weight, but the defaults are assigned to the `ActivityInstance.__schemaInstance` field, because the data needs to exist in order to be typechecked.

Reply:
Ohh interesting. I trust you know what you're doing here but it seems odd to have to generate literals to run type checking. In my mind type semantics exist for this purpose in the first place. I'm probably just confusing myself though, if you have time for a short walkthrough of this PR I'd be interested in joining!

Reply:
There's definitely something funky going on here that I don't understand, but see [this comment](https://github.com/NASA-AMMOS/aerie/pull/183#discussion_r882873669) for the current situation.